src/Database/Type/ColumnSchemaAwareInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+declare(strict_types=1);
+
+namespace Cake\Database\Type;
+
+use Cake\Database\DriverInterface;
+use Cake\Database\Schema\TableSchemaInterface;
+
+interface ColumnSchemaAwareInterface
+{
+    /**
+     * Generate the SQL fragment for a single column in a table.
+     *
+     * @param \Cake\Database\Schema\TableSchemaInterface $schema The table schema instance the column is in.
+     * @param string $column The name of the column.
+     * @param \Cake\Database\DriverInterface $driver The driver instance being used.
+     * @return string|null An SQL fragment, or `null` in case the column isn't processed by this type.
+     */
+    public function getColumnSql(TableSchemaInterface $schema, string $column, DriverInterface $driver): ?string;
+
+    /**
+     * Convert a SQL column definition to an abstract type definition.
+     *
+     * @param array $definition The column definition.
+     * @param \Cake\Database\DriverInterface $driver The driver instance being used.
+     * @return array|null Array of column information, or `null` in case the column isn't processed by this type.
+     */
+    public function convertColumnDefinition(array $definition, DriverInterface $driver): ?array;
```

Comment:
I'm concerned this is not well-defined. The user does not know what `$definition` contains even if it is driver-specific.

We also don't define what the expected result should be. Should the array contain all the keys that SchemaInterface::addColumn() accepts?

Reply:
> Should the array contain all the keys that SchemaInterface::addColumn() accepts?

Basically yes, according to the column type and driver.

Reply:
I think we need to document this as well as document what to expect from each driver. At the moment, driver-specific keys are an implementation detail that we are exposing.